rust/crd/src/s3logdir.rs

```diff
@@ -0,0 +1,170 @@
+                        match &server_verification.ca_cert {
+                            CaCert::WebPki {} => {}
+                            CaCert::SecretClass(_) => {
+                                return S3TlsCaVerificationNotSupportedSnafu.fail()
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        if let Some(conn) = s3bucket.as_ref().and_then(|i| i.connection.as_ref()) {
+            if conn.tls.as_ref().is_some() {
+                tracing::warn!("The resource indicates S3-access should use TLS: TLS-verification has not yet been implemented \
+            but an HTTPS-endpoint will be used!");
+            }
+        }
+        Ok(Some(S3LogDir {
+            bucket: s3bucket.unwrap(),
+            prefix,
+        }))
+    }
+
+    /// Constructs the properties needed for loading event logs from S3.
+    /// These properties are later written in the `HISTORY_CONFIG_FILE_NAME_FULL` file.
+    ///
+    /// The following properties related to credentials are not included:
+    /// * spark.hadoop.fs.s3a.aws.credentials.provider
+    /// * spark.hadoop.fs.s3a.access.key
+    /// * spark.hadoop.fs.s3a.secret.key
+    /// instead, the environment variables AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY are set
+    /// on the container start command.
+    pub fn history_server_spark_config(&self) -> BTreeMap<String, String> {
+        let mut result = BTreeMap::new();
+
+        result.insert("spark.history.fs.logDirectory".to_string(), self.url());
+
+        if let Some(endpoint) = self.bucket.endpoint() {
+            result.insert("spark.hadoop.fs.s3a.endpoint".to_string(), endpoint);
+        }
+
+        if let Some(conn) = self.bucket.connection.as_ref() {
+            if let Some(S3AccessStyle::Path) = conn.access_style {
+                result.insert(
+                    "spark.hadoop.fs.s3a.path.style.access".to_string(),
+                    "true".to_string(),
+                );
+            }
+        }
+        result
+    }
+
+    pub fn application_spark_config(&self) -> BTreeMap<String, String> {
+        let mut result = BTreeMap::new();
+        result.insert("spark.eventLog.enabled".to_string(), "true".to_string());
+        result.insert("spark.eventLog.dir".to_string(), self.url());
+        result
+    }
+
+    fn url(&self) -> String {
+        format!(
+            "s3a://{}/{}",
+            self.bucket.bucket_name.as_ref().unwrap().clone(), // this is guaranteed to exist at this point
+            self.prefix
+        )
+    }
+    pub fn credentials_volume(&self) -> Option<Volume> {
+        self.credentials()
+            .map(|credentials| credentials.to_volume(VOLUME_NAME_S3_CREDENTIALS))
+    }
+
+    pub fn credentials_volume_mount(&self) -> Option<VolumeMount> {
+        self.credentials().map(|_| VolumeMount {
+            name: VOLUME_NAME_S3_CREDENTIALS.into(),
+            mount_path: S3_SECRET_DIR_NAME.into(),
```

Comment:
Please keep in mind that we can have multiple s3 secrets (one for application and one for eventlogs). This is the reason i generally name volume name and volume mount path `<secret-class-name>-s3-credentials`

Reply:
Done.